### PR TITLE
fix: cache with specific python version

### LINKED ISSUE #6568: PR review companion failed to run
Recently, there are many `PR review companion` workflows which are failed to run.

It seems to be a `cache` problem. The `actions/cache` will cache `.local` with a specific key. But the `key` will be the same when the python version may be different. Consider those two action jobs:

- https://github.com/mdn/translated-content/runs/7088612871?check_suite_focus=true#step:5:6 (python 3.8.12)
- https://github.com/mdn/translated-content/runs/7091308837?check_suite_focus=true#step:5:6 (python 3.8.13)

With different python version but the `actions/cache` will restore the cache used in `python 3.8.12`.

And see snok/install-poetry#82. I think we should add `python version` to the `actions/cache` action's key to avoid this.

### PR DESCRIPTION
**Pull request by yin1999**
Fixes #6568

### CODE REVIEW

**Comment by yin1999** [comment]
Tested: https://github.com/yin1999/translated-content/runs/7103501129?check_suite_focus=true#step:6:4

The cache key will be specified with python version, and `snok/install-poetry` could find `python` now.

File: .github/workflows/pr-review-companion.yml [hunk 1]
```diff
@@ -76,10 +77,10 @@ jobs:
         uses: actions/cache@v3
         with:
           path: ~/.local
-          key: dotlocal-${{ runner.os }}-${{ hashFiles('.github/workflows/pr-review-companion.yml') }}
+          key: dotlocal-${{ runner.os }}-${{ hashFiles('.github/workflows/pr-review-companion.yml') }}-${{ steps.setup-python.outputs.python-version }}
```
Comment (yin1999) [inline .github/workflows/pr-review-companion.yml:80]: I'm not sure about `hashFiles('.github/workflows/pr-review-companion.yml')` it seems to be an empty string.

**Comment by yin1999** [comment]
pin @mdn/core-dev for a review

**Comment by yin1999** [comment]
Hi @schalkneethling, could you have a review on this. This problem is a bit annoying.

File: .github/workflows/pr-review-companion.yml (hunk 1 above)
Comment (LeoMcA) [reply .github/workflows/pr-review-companion.yml:80]: Yeah, we don't checkout the repo in this workflow, so there's no file to cache - I think that's ok, we can increment a number at the end of the cache key if we make any breaking changes. So, can you remove this part?

Comment (LeoMcA) [reply .github/workflows/pr-review-companion.yml:80]: We can remove it from the venv cache key below, as well. Perhaps we also want to include the python version in there, I don't think the poetry lockfile includes it.

**Review by LeoMcA — APPROVED** [review]
looks good if we remove those empty hashes, thanks!

we should make the same changes to this file in the `content` repo - would you like to open a similar PR there, or should I with the magic of [`Co-authored-by`](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors)?

File: .github/workflows/pr-review-companion.yml (hunk 1 above)
Comment (yin1999) [reply .github/workflows/pr-review-companion.yml]: thanks for answering this, I've add a trailing `0` with a comment to explain it.

**Comment by yin1999** [comment]
> looks good if we remove those empty hashes, thanks!
> 
> we should make the same changes to this file in the `content` repo - would you like to open a similar PR there, or should I with the magic of [`Co-authored-by`](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors)?

Ok, I'll make the same changes there.

**Review by LeoMcA — APPROVED** [review]

**Comment by LeoMcA** [comment]
Thank you!

**Pull request merged**